### PR TITLE
Founded partial recursion

### PR DESCRIPTION
**Pull request by sctfn**
This PR finishes up a long term project of mine- recursion on a general founded predicate. It’s a standard result in set theory textbooks, but we’ve never had it. 

While I was at it, I also proved a variant I’ve never seen in a textbook- you don’t need Infinity if you have transitivity. 

Given these two, should we plan on a move to main soon?

### CODE REVIEW

**Comment by tirix** [comment]
Welcome back! 
Looks great at first sight, I'll try to take more time to review in detail.

**Review by jkingdon — APPROVED** [review]
Putting this in a mathbox certainly looks good.

I don't really know enough to say one way or another about whether it belongs in main.

File: set.mm [hunk 1]
```diff
@@ -512104,7 +512129,7 @@ Real and complex numbers (cont.)
 
 $(
 =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-  Founded Recursion
+  Founded Parial Recursion
```
Comment (icecream17) [inline set.mm:512132]:
```suggestion
  Founded Partial Recursion
```

Comment (icecream17) [inline set.mm:512132]:
```suggestion
  Founded Partial Recursion
```

**Review by icecream17 — APPROVED** [review]
lgtm (oops accidentally deleted my previous comment) (other than typo)

File: set.mm (hunk 1 above)
Comment (tirix) [reply set.mm:512132]: Fixed.

**Review by tirix — APPROVED** [review]

**Pull request merged**